### ISSUE TITLE
Improve saving of custom player data

### ISSUE THREAD

**Issue by Maxopoly** [CivClassic/CivModCore#41]
Either save regularly (possibly locking issues with this approach) or switch over to fully SQL based approach

**Comment by Protonull** [comment]
Are you referring to player settings? (`/config`) or something else?

**Comment by Maxopoly** [comment]
Yea, that. I want to make it more stable when the server crashes.

**Comment by Protonull** [comment]
Would it be worth using [PersistentDataContainer](https://papermc.io/javadocs/paper/1.14/org/bukkit/persistence/PersistentDataContainer.html) instead of SQL'ing it?

**Comment by Maxopoly** [comment]
Idk, https://papermc.io/javadocs/paper/1.14/org/bukkit/persistence/PersistentDataType.html looks like an abomination

**Comment by Protonull** [comment]
I disagree, this looks far easier to deal with that the chunk meta stuff. Should we at least try using the native implementation before we add more SQL stuff?

**Comment by Maxopoly** [comment]
Don't really see the benefit.

Switching to this is basically a 100 % rewrite of all player data and its use cases.
My suggestion for this issue here would have been to only swap out the storage engine, which is not too hard to do

**Comment by Protonull** [comment]
Not necessarily, especially since the data is stored by plugin. There's easy way to do that in SQL sure, but not a good way.

**Comment by Protonull** [comment]
After looking into the source code of `PersistentDataContainer` and its surrounding classes, yeah it's garbage. It's more limited than the nms' NBT classes.. *so if we were to do this,* we'd might as well just use our `NBTCompound` class directly.
